solr/core/src/java/org/apache/solr/handler/admin/CoreAdminOperation.java

```diff
@@ -919,7 +928,93 @@ public void call(CallInfo callInfo) throws Exception {
         }
       }
     }
+  },
+  CREATESNAPSHOT_OP(CREATESNAPSHOT) {
+    @Override
+    public void call(CallInfo callInfo) throws Exception {
+      CoreContainer cc = callInfo.handler.getCoreContainer();
+      final SolrParams params = callInfo.req.getParams();
+
+      String commitName = params.required().get(CoreAdminParams.COMMIT_NAME);
+      String cname = params.required().get(CoreAdminParams.CORE);
+      try (SolrCore core = cc.getCore(cname)) {
+        if (core == null) {
+          throw new SolrException(ErrorCode.SERVER_ERROR, "Unable to locate core " + cname);
+        }
+        if (core.getDeletionPolicy().getLatestCommit() == null) {
```

Comment:
I see that ReplicationHandler is implemented differently. If the latestCommit is null, it uses the commit opened by the index searcher. Now I am using the same logic here, assuming the index searcher will always return non-null index commit.